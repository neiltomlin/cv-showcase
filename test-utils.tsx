import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import NeilCvTheme from "./styles/theme";

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={NeilCvTheme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactNode,
  options: RenderOptions<
    typeof import("@testing-library/dom/types/queries"),
    HTMLElement,
    HTMLElement
  > = {}
) => render(ui, { wrapper: AllTheProviders, ...options });
