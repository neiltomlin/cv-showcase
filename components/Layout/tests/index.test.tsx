import { screen } from "@testing-library/react";

import { Layout } from "../";
import { customRender } from "@/test-utils";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

test("Shows the Header and children", async () => {
  customRender(
    <Layout>
      <div>Test Content</div>
    </Layout>
  );

  expect(screen.getByText("Neil Tomlin - CV")).toBeInTheDocument();
  expect(screen.getByText("Test Content")).toBeInTheDocument();
});

test("Looks like it should do", async () => {
  const { asFragment } = customRender(
    <Layout>
      <div>Test Content</div>
    </Layout>
  );

  expect(asFragment()).toMatchSnapshot();
});
