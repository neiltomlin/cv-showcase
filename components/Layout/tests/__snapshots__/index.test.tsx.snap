// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Looks like it should do 1`] = `
<DocumentFragment>
  <header
    class="sc-bRKDuR kUfCqD"
  >
    <div
      class="sc-hvigdm kJZGKO"
    >
      <a
        class="sc-dTvVRJ eyTQkP"
        href="#main"
      >
        Skip to main content
      </a>
      <a
        href="/"
      >
        Neil Tomlin - CV
      </a>
      <div
        class="sc-fhHczv iRcZnd"
      >
        <a
          href="mailto:neiltomlin88@gmail.com"
        >
          neiltomlin88@gmail.com
        </a>
        <a
          href="tel:07866960512"
        >
          07866960512
        </a>
        <div
          class="sc-ggWZvA eCNiYi"
        >
          <a
            href="www.linkedin.com/in/neil-tomlin"
          >
            <img
              alt="LinkedIn"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
          </a>
          <a
            href="https://github.com/neiltomlin"
          >
            <img
              alt="GitHub"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
          </a>
        </div>
      </div>
    </div>
  </header>
  <div
    class="sc-jwTyAe fcnNLQ"
  >
    <div
      class="sc-hjsuWn ghDavz"
    >
      <div
        class="sc-jJLAfE jPTkuf"
      >
        <p>
          Site under construction
        </p>
        <img
          alt="Image showing a website being built"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
      <button>
        Close
      </button>
    </div>
  </div>
  <main
    class="sc-hwkwBN dQZWHR"
  >
    <div>
      Test Content
    </div>
  </main>
</DocumentFragment>
`;
